File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_lineal_regresion.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Thu Aug 13 20:07:46 2020

@author: zoro
"""

#Libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#import data set

dataset = pd.read_csv('Salary_Data.csv')
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values


#Split dataset in two subset 'Testing' and 'Training'
from sklearn.cross_validation import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 1/3,random_state = 0)

#Scaling values
"""from sklearn.preprocessing import StandardScaler
sc_x = StandardScaler()
x_train = sc_x.fit_transform(x_train)
x_test = sc_x.transform(x_test)
#"""

#Create a simple regression model for train subset
from sklearn.linear_model import LinearRegression
regression = LinearRegression()
regression.fit(x_train, y_train)

# Use the test subset
y_pred = regression.predict(x_test)

#creat a plot to see data train
plt.scatter(x_train, y_train, color="red")
plt.plot(x_train, regression.predict(x_train), color = "blue")
plt.title("Salary vs Time of Experience")
plt.xlabel("Time of Expereience")
plt.ylabel("Salary $")
plt.show()


#creat a plot to see data test
plt.scatter(x_test, y_test, color="red")
plt.plot(x_train, regression.predict(x_train), color = "blue")
plt.title("Salary vs Time of Experience")
plt.xlabel("Time of Expereience")
plt.ylabel("Salary $")
plt.show()

File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_lineal_regresion.R
#LINEAR REGRESSION SIMPLE
#Created on Thu Aug 13 20:07:46 2020

#@author: zoro
#

#PreProcessing dataset
# import dataset
dataset = read.csv('Salary_Data.csv')

#Split dataset in two subset 'Testing' and 'Training'
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3) #SplitRatio = TRAIN
training_set = subset(dataset, split == TRUE)
testing_set = subset(dataset, split == FALSE)

#scaling values
# training_set[,2:3] = scale(training_set[,2:3])
# testing_set[,2:3] = scale(testing_set[,2:3])

# Ajustar el modelo de regresion lineal simple con el conjunto de entrenamiento
#summary(regression)
regression = lm(formula = Salary ~ YearsExperience, training_set)

#Predecir resultados para el conjunto de testing
y_pred = predict(regression, newdata = testing_set)

#visualización datos
install.packages("ggplot2")
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary), colour= "red")+
  geom_line(aes(x = training_set$YearsExperience, y =  predict(regression, newdata = training_set)), color = "blue")+
  ggtitle("Salary vs Experience Years (Train)")+
  xlab("Year Experience")+
  ylab("Salary $")















